### PR TITLE
Fix function-no-unknown performance

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Ref #6869 

> Is there anything in the PR that needs further explanation?

Before:
Mean: 104.89167158823528 ms
Deviation: 9.601897806542654 ms

After:
Mean: 71.84089116666665 ms
Deviation: 10.215987936599932 ms

### CODE REVIEW

**Comment by jeddy3** [comment]
> LGTM! How about adding a changelog entry like this?

Oops, I'm rusty 😆 . I forgot to do that.

File: .changeset/shy-tools-rest.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+- Fixed: `function-no-unknown` performance
```
Comment (ybiquitous) [inline .changeset/shy-tools-rest.md:5]:
```suggestion
Fixed: `function-no-unknown` performance
```

**Review by ybiquitous — APPROVED** [review]
Thank you! 👍🏼

File: .changeset/shy-tools-rest.md (hunk 1 above)
Comment (jeddy3) [reply .changeset/shy-tools-rest.md]: 😅

**Pull request merged**